src/ai/backend/common/clients/valkey_client/client.py

```diff
@@ -592,3 +593,100 @@ async def wrapper(*args, **kwargs) -> R:
         return decorator
 
     return valkey_decorator
+
+
+def create_layer_aware_valkey_decorator_with_default(
+    layer: LayerType,
+):
+    """
+    Factory function to create layer-aware valkey decorators.
+
+    Args:
+        layer: The layer type for metric observation
+
+    Returns:
+        A valkey_decorator function configured for the specified layer
+    """
+
+    def valkey_decorator(
```

Comment:
I remember that we could just let it fail transparently when the Redis instance for log collection becomes unavailable, without impacting other parts of the system.
Is there any specialization on this part?